### PR TITLE
image: snap assertions into image

### PR DESCRIPTION
**Pull request by pedronis**
This fetches snap assertions at image construction time and make sure they are copied in the seed area of the image.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #1770, picking up on top of #1723 from mvo

**Comment by pedronis** [comment]
WIP

File: asserts/asserts.go [hunk 1]
```diff
@@ -94,6 +94,10 @@ type Ref struct {
 	PrimaryKey []string
 }
 
+func (ref *Ref) String() string {
```
Comment (mvo5) [inline asserts/asserts.go]: I like that!

File: tests/lib/fakestore/store/store.go [hunk 2]
```diff
@@ -49,16 +50,26 @@ type Store struct {
...
 }
 
-// NewStore creates a new store server serving snaps from the given top directory and assertions from topDir/asserts.
-func NewStore(topDir, addr string) *Store {
+// NewStore creates a new store server serving snaps from the given top directory and assertions from topDir/asserts. If assertFallback is true missing assertions are looked up in the main online store.
+func NewStore(topDir, addr string, assertFallback bool) *Store {
 	mux := http.NewServeMux()
+	var sto *store.Store
+	if assertFallback {
+		sto = store.New(nil, nil)
```
Comment (mvo5) [inline tests/lib/fakestore/store/store.go]: clever!

**Comment by mvo5** [comment]
Looks good, thanks for picking this up!

**Comment by chipaca** [comment]
Yeap, LGTM.

**Comment by mvo5** [comment]
Hm, `2016/08/30 09:46:42 Error preparing linode:ubuntu-core-16-64:tests/main/ : kill-timeout reached, cannot reconnect to linode:ubuntu-core-16-64 (Spread-G) after reboot: dial tcp 66.228.37.28:22: i/o timeout` - looks like we hit the reboot limit in linode again

**Pull request merged**